### DISCUSSION TITLE
WebHook conversations

### DISCUSSION THREAD

**Discussion by Rudique** [nutgram/nutgram#472] [Q&A] [answered]
I switched the bot to a web hook and the conversions stopped running, the documentation says to use a persistent cache adapter, advise me how to use it? 

use SergiX44\Nutgram\Conversations\Conversation;
use SergiX44\Nutgram\Nutgram;

class MyConversation extends Conversation {

    public function start(Nutgram $bot)
    {
        $bot->sendMessage('This is the first step!');
        $this->next('secondStep');
    }

    public function secondStep(Nutgram $bot)
    {
        $bot->sendMessage('Bye!');
        $this->end();
    }
}

Dialog hangs on the first step((

**Comment by martin-ro** [comment]
I have the same issue. Did you figure it out?

**Reply by Rudique** [reply]
Unfortunately no, conceptually I understand that I need to use Redis or a cache adapter to store the state and somehow register them in the bot, but nothing is written in the documentation and my timeline was burning, so I just run the bot on linux through systemd in longpolling mode and prescribed to bring it up every time it crashes. It's not a great solution, but I needed it to work.
If you figure it out, please let me know)

**Comment by Lukasss93** [comment, marked as answer]
- https://nutgram.dev/docs/configuration/cache

![image](https://github.com/nutgram/nutgram/assets/4071613/60693af2-bd6a-4557-958f-38a92e372a25)

- https://github.com/nutgram/nutgram/discussions/492#discussioncomment-6148833

**Reply by Rudique** [reply]
Thanks, I really should have read the answers more carefully, in my opinion it would be cool if the page 
https://nutgram.dev/docs/usage/conversations/
had a link to 
https://nutgram.dev/docs/configuration/cache/

And yes, thanks again)
